src/client/pages/NewAccountConsents.tsx

```diff
@@ -0,0 +1,129 @@
+import { SUPPORT_EMAIL } from '@/shared/model/Configuration';
+import { PasscodeErrors } from '@/shared/model/Errors';
+
+type NewAccountConsentsProps = RegistrationProps & {
+	geolocation?: GeoLocation;
+	appName?: AppName;
+	shortRequestId?: string;
+	pageError?: string;
+};
+
+const getErrorContext = (pageError?: string) => {
+	if (pageError === PasscodeErrors.PASSCODE_EXPIRED) {
+		return (
+			<>
+				<div>
+					Please request a new verification code to create your account.
+				</div>
+				<br />
+				<div>
+					If you are still having trouble, please contact our customer service
+					team at{' '}
+					<ThemedLink href={locations.SUPPORT_EMAIL_MAILTO}>
+						{SUPPORT_EMAIL}
+					</ThemedLink>
+					.
+				</div>
+			</>
+		);
+	}
+};
+
+export const NewAccountConsents = ({
+	email,
+	recaptchaSiteKey,
+	queryParams,
+	formError,
+	geolocation,
+	appName,
+	shortRequestId,
+	pageError,
+}: NewAccountConsentsProps) => {
+	const formTrackingName = 'register';
+
+	const formRef = useRef(null);
+	const formSubmitMethodRef = useRef(null);
+
+	usePageLoadOphanInteraction(formTrackingName);
+
+	const isJobs = queryParams.clientId === 'jobs';
+
+	const handleConsentChange = () => {
+		const formSubmitMethodInput =
+			formSubmitMethodRef.current as HTMLInputElement | null;
+		if (formSubmitMethodInput) {
+			// eslint-disable-next-line functional/immutable-data -- we need to mutate the object here to distinguish between the different methods of submittng the form
+			formSubmitMethodInput.value = 'toggle-action';
+		}
+
+		const formEl = formRef.current as HTMLFormElement | null;
+		formEl?.submit();
+	};
+
+	return (
+		<MinimalLayout
+			pageHeader="Complete your account"
+			shortRequestId={shortRequestId}
+			errorContext={getErrorContext(pageError)}
+			errorOverride={pageError}
+		>
+			<MainBodyText>
+				<strong>{email}</strong>
+			</MainBodyText>
+			<MainForm
+				formRef={formRef}
+				formAction={buildUrlWithQueryParams(
+					'/welcome/complete-account',
+					{},
+					queryParams,
+				)}
+				submitButtonText="Next"
+				recaptchaSiteKey={recaptchaSiteKey}
+				formTrackingName={formTrackingName}
+				disableOnSubmit
+				formErrorMessageFromParent={formError}
+				onSubmit={(e) => {
+					const formSubmitMethodInput =
+						formSubmitMethodRef.current as HTMLInputElement | null;
+					if (formSubmitMethodInput) {
+						// eslint-disable-next-line functional/immutable-data -- we need to mutate the object here to distinguish between the different methods of submittng the form
+						formSubmitMethodInput.value = 'submit-button';
+					}
+					registrationFormSubmitOphanTracking(e.target as HTMLFormElement);
+					return undefined;
+				}}
+				additionalTerms={[newsletterAdditionalTerms, isJobs === false].filter(
+					Boolean,
+				)}
+				shortRequestId={shortRequestId}
+			>
+				<input
+					type="hidden"
+					name="formSubmitMethod"
+					value="submit-button"
+					ref={formSubmitMethodRef}
+				/>
```

Comment:
i'm not quite understanding why this hidden form button is needed and it's relation to onSubmit?